subsys/bluetooth/host/scan.c

```diff
@@ -712,32 +713,36 @@ void bt_hci_le_per_adv_sync_established(struct net_buf *buf)
 	}
 
 	if (!pending_per_adv_sync ||
-	    (!atomic_test_bit(pending_per_adv_sync->flags,
-			      BT_PER_ADV_SYNC_SYNCING_USE_LIST) &&
+	    (!atomic_test_bit(pending_per_adv_sync->flags, BT_PER_ADV_SYNC_SYNCING_USE_LIST) &&
 	     ((pending_per_adv_sync->sid != evt->sid) ||
 	      bt_addr_le_cmp(&pending_per_adv_sync->addr, &evt->adv_addr)))) {
-		struct bt_le_per_adv_sync_term_info term_info;
-
 		BT_ERR("Unexpected per adv sync established event");
+		/* Request terminate of pending periodic advertising in controller */
 		per_adv_sync_terminate(sys_le16_to_cpu(evt->handle));
```

Comment:
Unrelated to your changes. Is it really needed to call _LE Periodic Advertising Terminate Sync_ in case the address is unexpected?

Reply:
I think that it is done to clean up controller periodic scanner, thanks to that host and controller are in the same state.

Reply:
This is the same behavior we also have for ACL connections

Reply:
> I think that it is done to clean up controller periodic scanner, thanks to that host and controller are in the same state.

@ppryga Hope controller does not depend on Host to call `per_adv_sync_terminate`, because the Controller shall cleanup its periodic sync before dispatching the sync established event with error code.

Reply:
@cvinayak my bad and thank you for the comment. 

There should not be call to `per_adv_sync_terminate` in case of sync timeout and sync with wrong CTE type.

I've just checked and both scenarios are not correctly handled in controller:
- sync with wrong CTE type does not clean everything up. It stops ticker but `sync` object is not released. Also `link_sync_lost` are not released.
- sync timeout is not handled correctly. It is handled through sync lost event instead sync established with error code. It looks like there is also missing release of `link_sync_estab` on sync lost execution path.

I'll fix `per_adv_sync_terminate`.

Reply:
@ppryga I have some fixes here, hope it covers the missing release of memory in zephyr controller. https://github.com/zephyrproject-rtos/zephyr/pull/40308